### PR TITLE
Query: Adds Full-Text Policy standard package and analyzer configuration support

### PR DESCRIPTION
**Pull request by adityasa**
## Description

This change adds preview support for the new full-text policy analyzer configuration fields, enabling users to specify:

- **Package type** (\legacy\ or \standard\) at the collection level
- **DefaultSpec** for inherited path configuration (language, tokenizer, filters, stop words)
- **Per-path overrides**: tokenizer, filters, stopWordListKind, addStopWords, removeStopWords

All new properties are preview-gated (\#if PREVIEW public #else internal\). The existing GA API surface is unchanged.

Based on backend design: CosmosDB commit 6dfb101c (Full-Text Policy analyzer configuration).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

- All existing FullText tests pass (unit + emulator)
- Added serialization tests for standard/legacy package round-trips
- Added fluent builder test for the new WithFullTextPolicy overload (preview-only)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextDefaultSpec.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    class FullTextDefaultSpec
+    {
+        /// <summary>
+        /// Gets or sets the default language locale (e.g., "en-US").
+        /// </summary>
+        [JsonProperty(PropertyName = "language", NullValueHandling = NullValueHandling.Ignore)]
+        public string Language { get; set; }
+
+        /// <summary>
+        /// Gets or sets the tokenizer method. Valid values: "word", "ngram", "ml".
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextDefaultSpec.cs]: >, "ngram", "ml" [](http://example.com/codeflow?start=67&length=15)

Remove. #Closed

File: Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextPath.cs [hunk 2]
```diff
@@ -45,6 +46,63 @@ public class FullTextPath : IEquatable<FullTextPath>
         [JsonProperty(PropertyName = "language", NullValueHandling = NullValueHandling.Ignore)]
         public string Language { get; set; }
 
+        /// <summary>
+        /// Gets or sets the tokenizer method. Valid values: "word", "ngram", "ml".
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextPath.cs]: >, "ngram", "ml" [](http://example.com/codeflow?start=67&length=15)

Remove #Closed

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/SettingsContractTests.cs [hunk 3]
```diff
@@ -1326,24 +1326,172 @@ public void FullTextPolicySerializationWithAllSupportedLanguages(string language
...
+                $"Serialized JSON should contain language: {language}");
+
+            FullTextPath deserialized = CosmosDeserialize<FullTextPath>(serialized);
+            Assert.IsNotNull(deserialized);
+            Assert.AreEqual("/testPath", deserialized.Path);
+            Assert.AreEqual(language, deserialized.Language,
+                $"Language mismatch after deserialization for: {language}");
+        }
+
+        [TestMethod]
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/SettingsContractTests.cs:1482]: Does this need to be guarded by PREVIEW check? #Closed

File: Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextDefaultSpec.cs (hunk 1 above)
Comment (adityasa) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextDefaultSpec.cs]: Fixed - removed 'ngram' and 'ml' from the doc comment since we're not shipping those tokenizer types in this iteration.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextPath.cs (hunk 2 above)
Comment (adityasa) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextPath.cs]: Fixed - same as above, removed the ngram/ml references.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/SettingsContractTests.cs (hunk 3 above)
Comment (adityasa) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/SettingsContractTests.cs:1482]: Good catch — wrapped all 3 new tests in \#if PREVIEW\ / \#endif\. They use the new preview-gated properties directly, so they should only compile in preview builds.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Fluent/ContainerSettingsTests.cs [hunk 4]
```diff
@@ -960,6 +961,97 @@ await databaseForFullTextSearch.DefineContainer(containerName, partitionKeyPath)
             }
         }
 
+#if PREVIEW
+        [TestMethod]
+        public async Task TestFullTextSearchPolicyStandardPackageWithQueryExecution()
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Fluent/ContainerSettingsTests.cs:967]: >TestFullTextSearchPolicyStandardPackageWithQueryExecution [](http://example.com/codeflow?start=26&length=57)

Can we validate all boundary conditions, update scenario and also invalid cases? #Closed

Comment (adityasa) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Fluent/ContainerSettingsTests.cs:967]: Added 4 additional emulator tests covering:

1. **StopWords round-trip + query** - validates \ddStopWords\/\
emoveStopWords\ persist and queries still execute correctly
2. **Update scenario** - creates container with one path, then adds a second via \ReplaceContainerAsync\, validates the policy is updated
3. **Per-path overrides** - tests multiple paths with different \	okenizer\, \ilters\, \stopWordListKind\ overrides, validates round-trip
4. **Invalid package** - verifies the service returns \BadRequest\ for an invalid package value (\\invalid_package\\)

All 5 tests pass against the emulator.

File: Microsoft.Azure.Cosmos/src/Fluent/Settings/FullTextPolicyDefinition.cs [hunk 5]
```diff
@@ -36,6 +40,30 @@ public FullTextPolicyDefinition(
             this.defaultLanguage = defaultLanguage;
         }
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/Fluent/Settings/FullTextPolicyDefinition.cs:47]: Can you chain the constructors delegating to the other one (if PREVIEW is defined) ? #Closed

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Fluent/ContainerSettingsTests.cs [hunk 6]
```diff
@@ -960,6 +961,654 @@ await databaseForFullTextSearch.DefineContainer(containerName, partitionKeyPath)
...
+        {
+            string fullTextPath1 = "/title", fullTextPath2 = "/description";
+            Database databaseForFullText = await this.GetClient().CreateDatabaseAsync("fullTextStandardDB",
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Fluent/ContainerSettingsTests.cs]: >"fullTextStandardDB", [](http://example.com/codeflow?start=86&length=21)

Here and in all places:
either all parameters on separate line, or all parameters on the same line. #Closed

**Comment by adityasa** [comment]
This change modifies public contract. You will need to run UpdateContracts.ps1 locally and checkin the contract changes. #Closed

File: Microsoft.Azure.Cosmos/src/Fluent/Settings/FullTextPolicyDefinition.cs (hunk 5 above)
Comment (adityasa) [reply Microsoft.Azure.Cosmos/src/Fluent/Settings/FullTextPolicyDefinition.cs:47]: Done — the base constructor now chains to the PREVIEW constructor via \: this(parent, package: null, defaultSpec: null, fullTextPaths, attachCallback)\ and just sets \defaultLanguage\ afterward.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Fluent/ContainerSettingsTests.cs (hunk 6 above)
Comment (adityasa) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/Fluent/ContainerSettingsTests.cs]: Fixed — all \CreateDatabaseAsync\ calls in the new tests now have parameters on a single line. Also added comprehensive validation tests for all filter/tokenizer/package/stopword combinations.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextDefaultSpec.cs [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+        /// Only applicable with the "standard" package and tokenizer "word".
+        /// </summary>
+        [JsonProperty(PropertyName = "filters", NullValueHandling = NullValueHandling.Ignore)]
+        public Collection<string> Filters { get; set; }
+
+        /// <summary>
+        /// Gets or sets the stop word list kind. Valid values: "none", "basic", "extended".
+        /// </summary>
+        [JsonProperty(PropertyName = "stopWordListKind", NullValueHandling = NullValueHandling.Ignore)]
+        public string StopWordListKind { get; set; }
```
Comment (Maya-Painter) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextDefaultSpec.cs:48]: Any reason this isn't an enum? It could be fine, just curious. In this case all spec validation is happening in the BE?  #Resolved

File: Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextPath.cs
Comment (kundadebdatta) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextPath.cs]: `FullTextPath.Equals` not updated for the 5 new fields (and no GetHashCode). Let's extend `Equals` to include the 5 new fields — using null-safe comparison and `SequenceEqual` for the `Collection<string>` fields. 

If possible, let's also add a matching `GetHashCode()`.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextDefaultSpec.cs (hunk 7 above)
Comment (adityasa) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextDefaultSpec.cs:48]: Exactly. Enums prevent independent changes in backend, so we follow the practice of using string.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextPath.cs
Comment (adityasa) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextPath.cs]: Done. Updated \Equals\ to include all 5 new fields (Tokenizer, Filters, StopWordListKind, AddStopWords, RemoveStopWords) with null-safe comparison and \SequenceEqual\ for collections. Added \GetHashCode\ override using Path, Language, Tokenizer, and StopWordListKind. Also added unit tests covering equality for each field difference, null collections, and hash code consistency.

**Review by Maya-Painter — DISMISSED** [review]

File: Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextPath.cs [hunk 8]
```diff
@@ -71,7 +130,43 @@ public void ValidateFullTextPath()
         /// <inheritdoc/>
         public bool Equals(FullTextPath that)
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextPath.cs]: Please also override
bool Equals(object other)
{
  return this.Equals(other as FullTextPath);
} #Closed

Comment (adityasa) [inline Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextPath.cs]: >that [](http://example.com/codeflow?start=40&length=4)

that -> other #Closed

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/SettingsContractTests.cs [hunk 9]
```diff
@@ -1326,26 +1326,310 @@ public void FullTextPolicySerializationWithAllSupportedLanguages(string language
...
+                Path = "/text",
+                Language = "en-US",
+                Tokenizer = "word",
+                StopWordListKind = "basic",
+            };
+            Assert.IsFalse(path1.Equals(pathNullCollections));
+        }
+#endif
+
+#if PREVIEW
```
Comment (adityasa) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/SettingsContractTests.cs]: Remove.

File: Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextPath.cs (hunk 8 above)
Comment (adityasa) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextPath.cs]: Done — added \override bool Equals(object obj)\ that delegates to \Equals(FullTextPath)\.

Comment (adityasa) [reply Microsoft.Azure.Cosmos/src/Resource/Settings/FullTextPath.cs]: Done — renamed \	hat\ to \other\.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/SettingsContractTests.cs (hunk 9 above)
Comment (adityasa) [reply Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.Tests/SettingsContractTests.cs]: Done — removed the redundant \#endif\ / \#if PREVIEW\ pair.

**Review by kundadebdatta — APPROVED** [review]
LGTM

**Review by sboshra — APPROVED** [review]
